tests/integration_tests/test_mlflow.py

```diff
@@ -0,0 +1,109 @@
+from mlflow.tracking import MlflowClient
+import py
+
+import optuna
+from optuna.integration.mlflow import MLflowCallback
+
+
+def _objective_func(trial, x_max=1.0):
+    # type: (optuna.trial.Trial, float) -> float
+
+    x = trial.suggest_uniform("x", -x_max, x_max)
+    y = trial.suggest_loguniform("y", 20, 30)
+    z = trial.suggest_categorical("z", (-1.0, 1.0))
+    assert isinstance(z, float)
+    trial.set_user_attr("my_user_attr", "my_user_attr_value")
+    return (x - 2) ** 2 + (y - 25) ** 2 + z
+
+
+def test_study_name(tmpdir):
```

Comment:
I'm a bit skeptical about using `py` which is in maintenance mode but it might be fine as we're using a `pytest` version that's meant to support it.

Reply:
Well, we need a temp directory to test without mocks. The type of this temp dir is in ˋpyˋ module. Nothing we can do here.